service/src/java/org/apache/hive/service/servlet/OTELExporter.java

```diff
@@ -0,0 +1,219 @@
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  public void exposeMetricsToOTEL() {
+    List<QueryInfo> liveQueries = operationManager.getLiveQueryInfos();
+    List<QueryInfo> historicalQueries = operationManager.getHistoricalQueryInfos();
+
+    LOG.debug("Found {} liveQueries and {} historicalQueries", liveQueries.size(), historicalQueries.size());
+
+    for (QueryInfo lQuery: liveQueries){
+      if(lQuery.getQueryDisplay() == null){
+        continue;
+      }
+      String queryID = lQuery.getQueryDisplay().getQueryId();
+      Span rootspan = queryIdToSpanMap.get(queryID);
+
+      //In case of live query previously encountered in past loops
+      if (rootspan != null) {
+        for (QueryDisplay.TaskDisplay task : lQuery.getQueryDisplay().getTaskDisplays()) {
+          if (task.getReturnValue() != null && task.getEndTime() != null
+                  && queryIdToTasksMap.get(queryID).add(task.getTaskId())) {
+            Context parentContext = Context.current().with(rootspan);
+            tracer.spanBuilder(queryID + " - " + task.getTaskId() + " - live")
+                    .setParent(parentContext).setAllAttributes(addTaskAttributes(task))
+                    .setStartTimestamp(task.getBeginTime(), TimeUnit.MILLISECONDS).startSpan()
+                    .end(task.getEndTime(), TimeUnit.MILLISECONDS);
+          }
+        }
+      } else {
+        // In case of live queries being seen for first time and has initialized its queryDisplay
+        rootspan = tracer.spanBuilder(queryID + " - live").startSpan();
+        Set<String> completedTasks = new HashSet<>();
+        Context parentContext = Context.current().with(rootspan);
+
+        Span initSpan = tracer.spanBuilder(queryID + " - live").setParent(parentContext).startSpan()
+                .setAttribute("QueryId", queryID)
+                .setAttribute("QueryString", lQuery.getQueryDisplay().getQueryString())
+                .setAttribute("BeginTime", lQuery.getBeginTime());
+        if (lQuery.getQueryDisplay().getErrorMessage() != null) {
+          initSpan.setAttribute("ErrorMessage", lQuery.getQueryDisplay().getErrorMessage());
+        }
+        initSpan.end();
+
+        for (QueryDisplay.TaskDisplay task : lQuery.getQueryDisplay().getTaskDisplays()) {
+          if (task.getReturnValue() != null && task.getEndTime() != null) {
+            completedTasks.add(task.getTaskId());
+            parentContext = Context.current().with(rootspan);
+            tracer.spanBuilder(queryID + " - " + task.getTaskId() + " - live")
+                    .setParent(parentContext).setAllAttributes(addTaskAttributes(task))
+                    .setStartTimestamp(task.getBeginTime(), TimeUnit.MILLISECONDS).startSpan()
+                    .end(task.getEndTime(), TimeUnit.MILLISECONDS);
+          }
+        }
+        queryIdToSpanMap.put(queryID,rootspan);
+        queryIdToTasksMap.put(queryID,completedTasks);
```

Comment:
this isn't formatted, can you format the code properly here & the other similar places